check_source.sh

```diff
@@ -10,6 +10,19 @@ if [ -f check-source.pl ] ; then
   perl check-source.pl || exit 1
 fi
 
+if [ -d .git ] ; then
+  echo "checking makefiles..."
+  ./updatemakes.sh
+  outofdate="$(git diff --name-only HEAD)"
```

Comment:
this makes the entire flow dependent if it's run from the git repo

as we're going to distribute archives as release, I think this needs another solution, but the idea is good!